storage/guarantees.go

```diff
@@ -7,9 +7,6 @@ import (
 // Guarantees represents persistent storage for collection guarantees.
 type Guarantees interface {
 
-	// Store inserts the collection guarantee.
-	Store(guarantee *flow.CollectionGuarantee) error
```

Comment:
## ⚠️ ❓ 

I am not sure I understand. A collection being guaranteed formally means that the collectors have finalized that collection -- that turns the pending collection to a "Guaranteed Collection". At the moment, the protocol is not fully implemented in this area. However, I think it is expected that Collector nodes would first store the collection when they propose it within their cluster. And later, once they finalize it, they add the `CollectionGuarantee` to their data base and send that guarantee to the Consensus nodes. 

You can compare your statement to ExecutionReceipts. Yes, consensus nodes generally handle execution receipts as part of the blocks. However, that does _not_ mean that Receipts "should [be] not stored alone" -- execution nodes do exactly that for a good reason. Exactly that same will likely hold true for collector nodes. 

**Please revert**

Reply:
> And later, once they finalize it, they add the CollectionGuarantee to their data base and send that guarantee to the Consensus nodes.

A collection guarantee is compound model derived from a cluster block, basically a cluster block QC and block payload. Collection Nodes store this data as HotStuff models (blocks, payloads, QCs), from which they can derive a collection guarantee at any time. 

In the same way that Consensus Nodes don't need to separately store a "finalized block" in their database when they finalize a block, Collection Nodes don't need to separately store a collection guarantee once it is guaranteed. 

In the context of the Consensus Follower, Leo is right that we do not want to store parts of a block independently; we store either the whole block and all its contents or nothing. So this change makes sense to me.

Reply:
Yeah, my point is that collection nodes currently don't store guarantee, but builds them on the fly (see collection finalizer) and submit it to consensus nodes (see collection pusher). And consensus nodes also don't store the guarantee when receiving them, but save them in the mempool ( see consensus ingestion core). 

The consensus nodes and other non-consensus nodes only store the guarantee when storing the block payload, which is implemented by Block.Store. That's why I removed the Store method on the Guarantees interface, for the same reason, I also removed `Payloads.Store`. This guarantees that if the database ever has a guarantee stored, then the block containing this guarantee must also exist in the database, since block and guarantee are stored in atomically, and this is the only way to store guarantee.